### PR TITLE
Fix passband_ripple() rejection-sampling budget crash

### LINKED ISSUE #482: PassbandRipple: passband_ripple() rejection-sampling budget too tight, crashes ~8% of the time at a valid parameter corner
## Summary

`torchsig/transforms/functional.py`'s `passband_ripple()` - used by the `PassbandRipple` impairment (fires on ~75% of signals at `Impairments(level=2)`, i.e. "impaired"/wireless-channel generation) - does rejection sampling: it repeatedly draws random filter taps until one meets a target ripple spec, giving up after a hardcoded 1000 tries and raising `ValueError("Passband ripple was unable to meet ripple specs.")`.

For `max_ripple_db`/`coefficient_decay_rate` values near the tight corner of `PassbandRipple`'s **own default distributions** (`max_ripple_db: (1, 2)`, `coefficient_decay_rate: (1, 5)`) combined with `num_taps=3` (from `num_taps: [2, 3]`), the per-draw acceptance probability is low enough (~0.25%, measured empirically) that there's roughly an **8% chance of exhausting the 1000-try budget on a single call**. Across a real dataset generation run (many thousands of `PassbandRipple` invocations via `DatasetCreator`), this reliably crashes the whole run with an uncaught exception in a `DataLoader` worker.

Reproduced this **deterministically, twice in a row**, generating an 18,000-sample impaired-level dataset with a fixed seed: both runs crashed in worker 7 at the same ~28%-through progress point.

## Reproduction

```python
import numpy as np
import torchsig.transforms.functional as F

rng = np.random.default_rng(12345)
fails = 0
trials = 200
for i in range(trials):
    data = (np.random.randn(4096) + 1j * np.random.randn(4096)).astype(np.complex64)
    try:
        F.passband_ripple(data, num_taps=3, max_ripple_db=1.0, coefficient_decay_rate=1.0, rng=rng)
    except ValueError:
        fails += 1
print(f"{fails}/{trials} failed")   # ~8% before the fix below, 0% after
```

Or via the empirical per-draw acceptance rate for that corner:

```python
import numpy as np

def acceptance_rate(num_taps, decay_rate, max_ripple_db, n_trials=200000, seed=1):
    rng = np.random.default_rng(seed)
    count = 0
    for _ in range(n_trials):
        gaussian = rng.normal(0, 1, num_taps) + 1j * rng.normal(0, 1, num_taps)
        decay = np.exp(-decay_rate * np.arange(0, num_taps))
        weights = gaussian * decay
        weights /= np.max(np.abs(weights))
        fft_db = 20 * np.log10(np.abs(np.fft.fftshift(np.fft.fft(weights, 1024))))
        ripple = np.max(fft_db) - np.min(fft_db)
        if ripple <= max_ripple_db:
            count += 1
    return count / n_trials

p = acceptance_rate(num_taps=3, decay_rate=1.0, max_ripple_db=1.0)
print(p, (1 - p) ** 1000)   # p ~= 0.0025, P(1000 consecutive fails) ~= 0.08
```

## Suggested fix

Two things:

1. Raise the rejection-sampling budget. 100,000 iterations reduces the failure probability to well under 1e-100 at the worst empirically-measured corner (`num_taps=3`, `max_ripple_db=1.0`, `coefficient_decay_rate=1.0`), while adding negligible runtime - each iteration is just a cheap 1024-point FFT.
2. Fix a related, currently-harmless-by-coincidence bug: the `while` loop's exit condition uses a hardcoded `1000` literal (`counter < 1000`) instead of the `max_counter` variable it's paired with in the post-loop check (`counter >= max_counter`). Today both happen to be 1000, so this has no visible effect, but bumping just the `max_counter` value (as in point 1) without also fixing this would silently disable the `ValueError` safety net instead of extending the retry budget - the loop would still bail out at 1000 tries and then proceed to apply the last (out-of-spec) `weights` without ever raising.

```python
    rng = rng or np.random.default_rng()
    # counter avoids infinite loop
    counter = 0
    max_counter = 100_000
    # initialize estimate such that it always enters loop
    estimate_ripple_db = max_ripple_db + 1

    while estimate_ripple_db > max_ripple_db and counter < max_counter:
        ...
```

I applied exactly this locally and confirmed it resolves the issue: 0/200 failures at the exact corner above (down from the expected ~8%), and a full-scale 18,000-sample impaired-variant generation that crashed twice pre-fix (same seed, same worker, same progress point both times) now completes cleanly.

Happy to open a PR with this change if useful - let me know.

## Environment

- torchsig 2.1.1 (installed via `pip install -e .` from a clone at commit `d9abfe1`/`9b1949e`)
- Python via conda env, Linux
- Found while generating data through the standard `TorchSigIterableDataset` + `WorkerSeedingDataLoader` + `DatasetCreator` pipeline with `Impairments(level=2)`, same pattern as the README/`scripts/generate_dataset_from_config.py`

### PR DESCRIPTION
**Pull request by sclkim1**
Fixes #482.

## Problem

`passband_ripple()` (used by the `PassbandRipple` impairment, which fires on ~75% of signals at `Impairments(level=2)`) does rejection sampling: it draws random filter taps until one meets a target ripple spec, giving up after a hardcoded 1000 tries and raising `ValueError`.

For `max_ripple_db`/`coefficient_decay_rate` values near the tight corner of `PassbandRipple`'s **own default distributions** (`max_ripple_db: (1, 2)`, `coefficient_decay_rate: (1, 5)`) combined with `num_taps=3` (from `num_taps: [2, 3]`), the per-draw acceptance probability is only ~0.25% (measured empirically) - about an 8% chance of exhausting the 1000-try budget on a single call. Across a real dataset generation run (many thousands of `PassbandRipple` invocations via `DatasetCreator`), this reliably crashes the whole run.

Reproduced deterministically, twice in a row, generating an 18,000-sample impaired-level dataset with a fixed seed: both runs crashed in the same `DataLoader` worker at the same ~28%-through progress point.

See #482 for the full writeup (root-cause derivation, acceptance-probability math).

## Fix

Two changes in `passband_ripple()`:

1. Raise the rejection-sampling budget from 1000 to 100,000 tries. This reduces the failure probability to well under 1e-100 at the worst empirically-measured corner, while adding negligible runtime (each try is a cheap 1024-point FFT).
2. Fix the loop's exit condition, which used a hardcoded `1000` literal (`counter < 1000`) instead of the `max_counter` variable it's paired with in the post-loop check (`counter >= max_counter`). Today both happen to be 1000, so this had no visible effect - but bumping just `max_counter` without also fixing this would have silently disabled the `ValueError` safety net instead of extending the budget (the loop would still bail at 1000 tries, then proceed to apply the last, out-of-spec `weights` without ever raising).

## Testing

- Added `test_passband_ripple_worst_case_corner_does_not_crash` (`tests/transforms/test_functional.py`): runs the worst-case corner (`num_taps=3`, `max_ripple_db=1.0`, `coefficient_decay_rate=1.0`) across 200 independent seeds and asserts no `ValueError`. Passes locally after the fix (existing `test_passband_ripple`/`test_PassbandRipple` tests also still pass).
- Manually verified: 0/200 failures at the exact worst-case corner via a standalone repro (down from the expected ~8% pre-fix).
- Regenerated the real 18,000-sample impaired-level dataset that crashed twice pre-fix; it now completes cleanly end-to-end.

### CODE REVIEW

**Comment by sclkim1** [comment]
Closing this PR - not looking to carry changes in TorchDSP/torchsig right now. The underlying bug report stays open at #482 in case it's useful; the fix is available there if anyone wants to pick it up.
